src/unistd.rs

```diff
@@ -112,6 +114,50 @@ pub fn chdir<P: ?Sized + NixPath>(path: &P) -> Result<()> {
 }
 
 #[inline]
+pub fn mkdir<P: ?Sized + NixPath>(path: &P, mode: Mode) -> Result<()> {
+    let res = try!(path.with_nix_path(|cstr| {
+        unsafe { libc::mkdir(cstr.as_ptr(), mode.bits() as mode_t) }
+    }));
+
+    Errno::result(res).map(drop)
+}
+
+#[inline]
+pub fn getcwd() -> Result<PathBuf> {
+    let mut buf = Vec::with_capacity(512);
+    loop {
+        unsafe {
+            let ptr = buf.as_mut_ptr() as *mut libc::c_char;
+
+            // The buffer must be large enough to store the absolute pathname plus
+            // a terminating null byte, or else null is returned.
+            // To safely handle this we start with a reasonable size (512 bytes)
+            // and double the buffer size upon every error
+            if !libc::getcwd(ptr, buf.capacity()).is_null() {
+                let len = CStr::from_ptr(ptr).to_bytes().len();
+                buf.set_len(len);
+                buf.shrink_to_fit();
+                let s = try!(CString::new(buf).map_err(|_| Error::Sys(Errno::EILSEQ)));
+                let s = try!(s.into_string().map_err(|_| Error::Sys(Errno::EILSEQ)));
+                return Ok(PathBuf::from(&s));
+            } else {
+                let error = Errno::last();
+                // ERANGE means buffer was too small to store directory name
+                if error != Errno::ERANGE {
+                    return Err(Error::Sys(error));
+                }                
```

Comment:
Thanks for the hint. I've corrected those now. Any idea why `cargo clippy` did not warn about those?